internal/archive/export_test.go

```diff
@@ -14,3 +14,12 @@ func FakeDo(do func(req *http.Request) (*http.Response, error)) (restore func())
 		bulkDo = _bulkDo
 	}
 }
+
+type Credentials = credentials
+var FindCredentials = findCredentials
+func FakeCredentialsDir(credentialsDir string) (restore func()) {
+	testCredentialsDir = credentialsDir
+	return func() {
+		testCredentialsDir = ""
+	}
```

Comment:
This needs to be
```
func FakeCredentialsDir(dir string) (restore func()) {
	oldCredentialsDir := credentialsDir
        credentiasDir = dir
	return func() {
		credentialsDir = oldCredentialsDir
	}
}
```
In other words:
1. No "test" prefix on the global, as it's the actual value used rather than a test value
2. The faking function stores and restores the previous value, instead of pretending to know what was there before

Reply:
Commented below on usage. I've removed the fake function and test `FindCredentialsInDir` directly.